### ISSUE TITLE
✨ 😅  Is possibale to use the ChatGPT of OpenAI to train this ChatGPT?

### ISSUE THREAD

**Issue by Yonv1943** [lucidrains/PaLM-rlhf-pytorch#23]
OpenAI used **40 people** when training their own chatGPT, and the annotation process lasted for **3 months**.

It is difficult for our open source community （github） to reproduce the **Reinforcement Learning by Human Feedback (RLHF)** for this work, as OpenAI also employs 40 people to complete human feedback. 

However, we can **treat OpenAI's web version of chatGPT as human, who can annotate data ✨  for us** when training our own chatGPT. 

> Step 2, A labeler (human or OpenAI chatGPT) ranks the outputs from best to worst.

![chatgpt.png](https://github.com/lucidrains/PaLM-rlhf-pytorch/blob/main/chatgpt.png?raw=true)

This sounds a bit funny😅, but I currently think it's doable.
@lucidrains

**Comment by draganjovanovich** [comment]
Its forbidden by the ToS.

**Comment by Yonv1943** [comment]
You means that "it is forbidden by the Terms of Service (ToS) of OpenAI ChatGPT".

Thank you for your response to my question. I will close this issue.

**Comment by zzzacwork** [comment]
Can you point me to where they have those results? I only saw the paper

**Comment by mqurishi** [comment]
Please check this
https://github.com/LAION-AI/Open-Assistant

**Comment by zzzacwork** [comment]
Thanks a lot!

**Comment by Yonv1943** [comment]
It is possibale to use the ChatGPT of OpenAI to train our own ChatGPT.

> The figure below illustrates how we obtained the Alpaca model. For the data, we generated instruction-following demonstrations by building upon the self-instruct method. We started with the 175 human-written instruction-output pairs from the [self-instruct seed set](https://github.com/yizhongw/self-instruct). We then prompted text-davinci-003 to generate more instructions using the seed set as in-context examples. We improved over the self-instruct method by simplifying the generation pipeline (see details in [GitHub](https://github.com/tatsu-lab/stanford_alpaca#data-generation-process)) and significantly reduced the cost. Our data generation process results in 52K unique instructions and the corresponding outputs, which **costed less than $500 using the OpenAI API.**

![](https://crfm.stanford.edu/static/img/posts/2023-03-13-alpaca/alpaca_main.jpg)

https://crfm.stanford.edu/2023/03/13/alpaca.html

https://github.com/tatsu-lab/stanford_alpaca

**Comment by rami2102** [comment]
> It is possibale to use the ChatGPT of OpenAI to train our own ChatGPT.
> 
> > The figure below illustrates how we obtained the Alpaca model. For the data, we generated instruction-following demonstrations by building upon the self-instruct method. We started with the 175 human-written instruction-output pairs from the [self-instruct seed set](https://github.com/yizhongw/self-instruct). We then prompted text-davinci-003 to generate more instructions using the seed set as in-context examples. We improved over the self-instruct method by simplifying the generation pipeline (see details in [GitHub](https://github.com/tatsu-lab/stanford_alpaca#data-generation-process)) and significantly reduced the cost. Our data generation process results in 52K unique instructions and the corresponding outputs, which **costed less than $500 using the OpenAI API.**
> 
> ![](https://camo.githubusercontent.com/1bf77dd184a6fa2af989682873e44685ab1b899a586276c62aa622307559b506/68747470733a2f2f6372666d2e7374616e666f72642e6564752f7374617469632f696d672f706f7374732f323032332d30332d31332d616c706163612f616c706163615f6d61696e2e6a7067)
> 
> https://crfm.stanford.edu/2023/03/13/alpaca.html
> 
> https://github.com/tatsu-lab/stanford_alpaca

Thanks a lot for your insightful sharing :)

Can you explain please how the method you used for training is compatible with ChatGPT and LLAMA 2 ToS?
